drivers/adc/adc_stm32.c

```diff
@@ -567,7 +567,8 @@ static int adc_stm32_calibrate(const struct device *dev)
 	adc_stm32_calibration_start(dev);
 #endif /* DT_HAS_COMPAT_STATUS_OKAY(st_stm32f1_adc) */
 
-#ifdef CONFIG_SOC_SERIES_STM32H7X
+#if defined(CONFIG_SOC_SERIES_STM32H7X) || \
+	defined(CONFIG_CPU_CORTEX_M7)
```

Comment:
If youre point is to prevent M4 H7 access to this code, it should rather be:
```suggestion
#if defined(CONFIG_SOC_SERIES_STM32H7X) && \
	defined(CONFIG_CPU_CORTEX_M7)
```

Reply:
Thanks!